pusher-api/user-guides/create-event-subscriptions.md

```diff
@@ -0,0 +1,175 @@
+       {
+         "uuid": "ef64c5e2-4e16-11e8-9c2d-fa7ae01bbebc",
+         "transportName": "WEBHOOK",
+         "eventNames": [
+            "ProductCreated",
+            "PurchaseCreated"
+         ],
+         "destination": "https://yoururl.domain",
+         "contactEmail": "email_if_it_breaks@domain.com"
+       }   
+    ```
+    
+2. Check that the response returns with an HTTP status code `200 OK`.
+    * If yes, the subscription is created successfully.
+    * If no, update the `POST` request according to the error message. For more information on error messages, see [HTTP status code](../api-reference.md#createHttpStatusCode).
+    
+3. Save the value of `signingKey` from the response. This is the key used to sign all requests and should be stored so that you can validate the request. 
+
+    ```json
+    {
+        ...
+        "signingKey": "G6MWAEw1Fc6FWPkfiJiZ3j8Ya76I5ZbEDVPtzcPl6L6scsylmK5AEDyNyMe8N5cy"
+      }
+    ```
+4. If you use the destination URL to receive events for the first time, check that the server receives a test message.
+
+    Example:
+    
+    The example shows a test message.
+    ```json
+    {
+      "eventName" : "TestMessage",
+      "organizationUuid" : "ef64c5e2-4e16-11e8-9c2d-fa7ae01bbebc",
+      "messageId" : "0f674460-fab5-11e7-a310-0002ebd6a43c",
+      "payload" : { "data" : "payload" }
+    }
+    ```
+
+## Step 4: Set up verification for events origin
+After event subscriptions are created, you need to set up a mechanism for verifying that events come from Zettle by checking the signature in the events. 
+
+The signature hash is generated as hexdigest using HMAC with SHA-256 as the cryptographic hash function. 
+
+To verify that events come from Zettle, calculate a signature and compare it with the value in the HTTP header `X-iZettle-Signature` of the incoming events. 
+
+1. Calculate a signature by concatenating the timestamp and payload of the incoming event `.`: `<timestamp>.<payload>` and using the stored signing key that you stored in [Step 3: Create an event subscription](#step-3-create-an-event-subscription).
+
+    The following examples uses Python, PHP, and Java for calculating a signature. 
+
+     <!-- what's the prerequisite for using the code? -->
+
+    <details>
+      <summary>Click to see a Python 2 example.</summary>
+        
+      ```python
+        import hmac
+        import hashlib
+        ...
+        payload_to_sign = '{}.{}'.format(timestamp, payload)
+        signature = hmac.new(bytes(signing_key), msg = bytes(payload_to_sign), digestmod = hashlib.sha256).hexdigest()       
+      ```
+       
+    </details>
+       
+    <details>
+      <summary>Click to see a Python 3 example.</summary>
+           
+      ```python
+        import hmac
+        import hashlib
+        ...
+        payload_to_sign = '{}.{}'.format(timestamp, payload)
+        signature = hmac.new(bytes(signing_key, 'UTF-8'), msg = bytes(payload
```

Comment:
The code has been clipped here, this is the complete code snippet:

```
import hmac
import hashlib
...
payload_to_sign = '{}.{}'.format(timestamp, payload)
signature = hmac.new(bytes(signing_key, 'UTF-8'), msg = bytes(payload_to_sign, 'UTF-8'), digestmod = hashlib.sha256).hexdigest()
```

Reply:
Thanks Pär! I will fix it.